lxd/container_lxc.go

```diff
@@ -3406,9 +3406,20 @@ func (c *containerLXC) Migrate(cmd uint, stateDir string, function string, stop
 		 * namespace.
 		 */
 		if !c.IsPrivileged() {
-			if err := c.IdmapSet().ShiftRootfs(stateDir); err != nil {
+			err = c.StorageStart()
+			if err != nil {
 				return err
 			}
+
+			err = c.IdmapSet().ShiftRootfs(stateDir)
+			err2 := c.StorageStop()
+			if err != nil {
+				return err
+			}
+
+			if err2 != nil {
+				return err2
+			}
```

Comment:
I'm assuming the intent was for those two if statements to be the other way around, so always calling StorageStop.